examples/thunderstorm_detection_and_tracking.py

```diff
@@ -0,0 +1,199 @@
+from pysteps.utils import to_reflectivity
+from pysteps.visualization import plot_precip_field, plot_track, plot_cart_contour
+
+################################################################################
+# Read the radar input images
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# A series of 20 files containing Swiss Cartesian gridded rainrate are imported. Since the
+# algorithm is tuned to Swiss max-reflectivity data, the rainrate is transformed to
+# reflectivity.
+
+date = datetime.strptime("201607112100", "%Y%m%d%H%M")
+data_source = rcparams.data_sources["mch"]
+
+root_path = data_source["root_path"]
+path_fmt = data_source["path_fmt"]
+fn_pattern = data_source["fn_pattern"]
+fn_ext = data_source["fn_ext"]
+importer_name = data_source["importer"]
+importer_kwargs = data_source["importer_kwargs"]
+timestep = data_source["timestep"]
+
+###############################################################################
+# Load the data from the archive
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+fns = io.archive.find_by_date(
+    date, root_path, path_fmt, fn_pattern, fn_ext, timestep, num_next_files=20
+)
+
+importer = io.get_method(importer_name, "importer")
+R, _, metadata = io.read_timeseries(fns, importer, **importer_kwargs)
+Z, metadata = to_reflectivity(R, metadata)
+timelist = metadata["timestamps"]
+
+pprint(metadata)
+
+###############################################################################
+# Example of thunderstorm identification in a single timestep.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# The function tstorm_detect.detection requires a 2-D input image, all further inputs are
+# optional.
+
+
+input_image = Z[2, :, :].copy()
+time = timelist[2]
+cells_id, labels = tstorm_detect.detection(
+    input_image,
+    dyn_thresh=True,
+    time=time,
+)
+
+###############################################################################
+# Example of thunderstorm tracking over a timeseries.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# The tstorm-dating function requires the entire pre-loaded time series.
+# The first two timesteps are required to initialize the
+# flow prediction and are not used to compute tracks.
+
+track_list, cell_list, label_list = tstorm_dating.dating(
+    input_video=Z,
+    timelist=timelist,
+    dyn_thresh=False,
+)
+
+###############################################################################
+# Plotting the results
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Plot precipitation field
+plot_precip_field(Z[2, :, :], geodata=metadata, units=metadata["unit"])
+
+# Add the identified cells
+plot_cart_contour(cells_id.cont, geodata=metadata)
+
+# Filter the tracks to only contain cells existing in this timestep
+
+IDs=cells_id.ID.values
+track_filt=[]
+for track in track_list:
+    if np.unique(track.ID) in IDs: track_filt.append(track)
+
+# Add their tracks
+plot_track(track_filt, geodata=metadata)
+plt.show()
+
+#%%
+################################################################################
+# Example with US MRMS data
```

Comment:
This is a great idea, @feldmann-m , but unfortunately loading the mrms data requires the pygrib package which is a bit problematic to install without conda (which we don't use on readthedocs). Also, the US data are somewhat heavy and might slow down the build of the documentation or event stop it due to time or memory limits.

I suggest that you only use Swiss data in this example, while the mrms dataset could become a more in-depth tutorial that we can publish as a Google Colab notebook. What do you think?